explore/marginal-gain/2.Marginal-Gain-Multiple-Mutations.py

```diff
@@ -0,0 +1,185 @@
+    """
+    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.1, random_state=0)
+    pipeline.fit(X=X_train, y=y_train)
+    y_pred_train = pipeline.decision_function(X_train)
+    y_pred_test = pipeline.decision_function(X_test)
+    cv_score_df = grid_scores_to_df(pipeline.grid_scores_)
+    series['mean_cv_auroc'] = cv_score_df.score.max()
+    series['training_auroc'] = roc_auc_score(y_train, y_pred_train)
+    series['testing_auroc'] = roc_auc_score(y_test, y_pred_test)
+    return series
+
+def grid_scores_to_df(grid_scores):
+    """
+    Convert a sklearn.grid_search.GridSearchCV.grid_scores_ attribute to 
+    a tidy pandas DataFrame where each row is a hyperparameter-fold combinatination.
+    """
+    rows = list()
+    for grid_score in grid_scores:
+        for fold, score in enumerate(grid_score.cv_validation_scores):
+            row = grid_score.parameters.copy()
+            row['fold'] = fold
+            row['score'] = score
+            rows.append(row)
+    df = pd.DataFrame(rows)
+    return df
+
+auroc_df = {}
+
+
+# In[46]:
+
+get_ipython().run_cell_magic('time', '', '# Train model a: covariates only.\nwarnings.filterwarnings("ignore") # ignore deprecation warning for grid_scores_\nrows = list()\nfor m in list(mutations):\n    series = pd.Series()\n    series[\'mutation\'] = m\n    series[\'symbol\'] = mutations[m]\n    rows.append(get_aurocs(X[\'a\'], Y[m], cv_pipeline[\'a\'], series))\nauroc_df[\'a\'] = pd.DataFrame(rows)\nauroc_df[\'a\'].sort_values([\'symbol\', \'testing_auroc\'], ascending=[True, False], inplace=True)')
+
+
+# In[47]:
+
+get_ipython().run_cell_magic('time', '', '# Train model b: covariates with gene expression data.\nwarnings.filterwarnings("ignore") # ignore deprecation warning for grid_scores_\nrows = list()\nfor m in list(mutations):\n    series = pd.Series()\n    series[\'mutation\'] = m\n    series[\'symbol\'] = mutations[m]\n    rows.append(get_aurocs(X[\'b\'], Y[m], cv_pipeline[\'b\'], series))\nauroc_df[\'b\'] = pd.DataFrame(rows)\nauroc_df[\'b\'].sort_values([\'symbol\', \'testing_auroc\'], ascending=[True, False], inplace=True)')
+
+
+# In[50]:
+
+display(auroc_df['a'])
+display(auroc_df['b'])
+
+
+# In[65]:
+
+auroc_df['a'].to_csv('auroc_covariates_only.tsv', index=False, sep='\t', float_format='%.5g')
+auroc_df['b'].to_csv('auroc_covariates_and_expression.tsv', index=False, sep='\t', float_format='%.5g')
+auroc_df['c'] = auroc_df['b'].loc[:,'mean_cv_auroc':]-auroc_df['a'].loc[:,'mean_cv_auroc':]
+auroc_df['c'][['mutation', 'symbol']] = auroc_df['b'].loc[:, ['mutation', 'symbol']]
+auroc_df['c']
+
+
+# # Covariates only vs covariates+expression model
+
+# In[68]:
+
+plot_df = pd.melt(auroc_df['c'], id_vars='symbol', value_vars=['mean_cv_auroc', 'training_auroc', 'testing_auroc'], var_name='kind', value_name='auroc')
+grid = sns.factorplot(y='symbol', x='auroc', hue='kind', data=plot_df, kind="bar")
```

Comment:
Can you change the x-axis label from `auroc` to `Δ AUROC`, so it's clear that it's the change in AUROC?